pkg/archive/archive.go

```diff
@@ -486,16 +486,17 @@ func (ta *tarAppender) addTarFile(path, name string) error {
 		if err != nil {
 			return err
 		}
-		hdr.Uid, hdr.Gid, err = ta.IDMappings.ToContainer(fileIDPair)
+		uid, gid, err := ta.IDMappings.ToContainer(fileIDPair)
 		if err != nil {
 			return err
 		}
+		hdr.Uid, hdr.Gid = int(uid), int(gid)
```

Comment:
We seem to do a lot of conversion from int to uint32 for hdr.Uid and hdr.Gid.  Would it be easier/better to convert those two fields to uint32 and then change them to convert them to int if necessary?  I may have missed a use point that makes that problematic.

Reply:
one problem I see is that `os.Chown` uses `int`, and on i386 an `int` cannot represent an `uint32` as they are both 32 bits.

While `uint32` is more correctly mapping what the kernel does, converting to `uint32` won't still solve this problem so we are back to square one to the initial problem.  We cannot use the full 0-4294967295 range on i386.

Reply:
So, we should just close this and go with the fix on CRI-O?

Reply:
I think so, let's not add more logic to handle uids>=2147483647 on 386.  Fixing it here could enable to have these uids mapped in the userNS but these uids could not be used to chown files but I don't think it is worth.